pkg/capture/capture_test.go

```diff
@@ -256,7 +257,8 @@ func testDeadlockHighTraffic(t *testing.T) {
 	for mockSrc.CanAddPackets() {
 		require.Nil(t, mockSrc.AddPacket(testPacket))
 	}
-	errChan := mockSrc.Run(time.Microsecond)
+	errChan, err := mockSrc.Run(time.Microsecond)
```

Comment:
I didn't fully grasp why there's an error channel and an error returned for the `Run` method.

Reply:
Because the `err` is for immediately failing during `Run()` (because the user may not proceed in that case), the `errChan` is for any errors that may (or may not) occur during the actual processing of packets. This is for the safeguard of the no-drain source (ensuring that the ring buffer is populated when starting `Run()`.